tests/tests_integration/test_api/test_datapoints.py

```diff
@@ -577,6 +591,51 @@ def test_retrieve_nothing__multiple(
                     assert isinstance(r.is_step, bool)
                     assert isinstance(r.is_string, bool)
 
+    @pytest.mark.parametrize(
+        "retrieve_method_name, kwargs",
+        itertools.product(
+            ["retrieve", "retrieve_arrays", "retrieve_dataframe"],
+            [dict(target_unit="temperature:deg_f"), dict(target_unit_system="Imperial")],
+        ),
+    )
+    def test_retrieve_methods_in_target_unit(
+        self,
+        retrieve_method_name: str,
+        kwargs: dict,
+        cognite_client: CogniteClient,
+        timeseries_degree_c_minus40_0_100: TimeSeries,
+    ) -> None:
+        timeseries = timeseries_degree_c_minus40_0_100
+        retrieve_method = getattr(cognite_client.time_series.data, retrieve_method_name)
+
+        res = retrieve_method(external_id=timeseries.external_id, end=3, **kwargs)
+
+        if isinstance(res, pd.DataFrame):
+            res = DatapointsArray(value=res.values)
+
+        assert math.isclose(res.value[0], -40, abs_tol=0.5)
+        assert math.isclose(res.value[1], 32, abs_tol=0.5)
+        assert math.isclose(res.value[2], 212, abs_tol=0.5)
```

Comment:
I would prefer going with the defaults, unless we suspect a larger relative error than 1e-9 (the default), which I don't see for a multiply+add operation.
```suggestion
        assert math.isclose(res.value[0], -40)
        assert math.isclose(res.value[1], 32)
        assert math.isclose(res.value[2], 212)
```